RecoBTag/MXNet/plugins/ParticleNetJetTagsProducer.cc

```diff
@@ -0,0 +1,287 @@
+    input_shapes_.push_back(group_pset.getParameter<std::vector<unsigned>>("input_shape"));
+    auto &prep_params = prep_info_map_[group_name];
+    prep_params.var_length = group_pset.getParameter<unsigned>("var_length");
+    prep_params.var_names = group_pset.getParameter<std::vector<std::string>>("var_names");
+    const auto &var_info_pset = group_pset.getParameterSet("var_infos");
+    for (const auto &var_name : prep_params.var_names) {
+      const auto &var_pset = var_info_pset.getParameterSet(var_name);
+      double median = var_pset.getParameter<double>("median");
+      double norm_factor = var_pset.getParameter<double>("norm_factor");
+      prep_params.var_info_map[var_name] = PreprocessParams::VarInfo(median, norm_factor);
+    }
+
+    // create data storage with a fixed size vector initilized w/ 0
+    unsigned len = prep_params.var_length * prep_params.var_names.size();
+    data_.emplace_back(len, 0);
+  }
+
+  if (debug_) {
+    for (unsigned i = 0; i < input_names_.size(); ++i) {
+      const auto &group_name = input_names_.at(i);
+      std::cout << group_name << "\nshapes: ";
+      for (const auto &x : input_shapes_.at(i)) {
+        std::cout << x << ", ";
+      }
+      std::cout << "\nvariables: ";
+      for (const auto &x : prep_info_map_.at(group_name).var_names) {
+        std::cout << x << ", ";
+      }
+      std::cout << "\n";
+    }
+  }
+
+  // init MXNetPredictor
+  predictor_.reset(new mxnet::cpp::Predictor(*cache->block));
+  predictor_->set_input_shapes(input_names_, input_shapes_);
+
+  // get output names from flav_names
+  for (const auto &flav_name : flav_names_) {
+    produces<JetTagCollection>(flav_name);
+  }
+}
+
+ParticleNetJetTagsProducer::~ParticleNetJetTagsProducer() {}
+
+void ParticleNetJetTagsProducer::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+  // pfParticleNetJetTags
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("src", edm::InputTag("pfParticleNetTagInfos"));
+  edm::ParameterSetDescription preprocessParams;
+  preprocessParams.setAllowAnything();
+  desc.add<edm::ParameterSetDescription>("preprocessParams", preprocessParams);
+  desc.add<edm::FileInPath>(
+      "model_path", edm::FileInPath("RecoBTag/Combined/data/ParticleNetAK8/General/V00/ParticleNet-symbol.json"));
+  desc.add<edm::FileInPath>(
+      "param_path", edm::FileInPath("RecoBTag/Combined/data/ParticleNetAK8/General/V00/ParticleNet-0000.params"));
+  desc.add<std::vector<std::string>>(
+      "flav_names",
+      std::vector<std::string>{
+          "probTbcq",  "probTbqq",  "probTbc",   "probTbq",  "probTbel", "probTbmu",      "probTbta",
+          "probWcq",   "probWqq",   "probZbb",   "probZcc",  "probZqq",  "probHbb",       "probHcc",
+          "probHqqqq", "probQCDbb", "probQCDcc", "probQCDb", "probQCDc", "probQCDothers",
+      });
+  desc.addOptionalUntracked<bool>("debugMode", false);
+
+  descriptions.add("pfParticleNetJetTags", desc);
```

Comment:
it looks like this is cloned later in pfParticleNet_cff to an item with the same name.
It would be safer to have a more "default" (not-to-be-used directly) name.
```descriptions.addWithDefaultLabel(desc);``` would work and make particleNetJetTagsProducer

Reply:
Changed to use `descriptions.addWithDefaultLabel(desc);`.